framework/src/geomsearch/NearestNodeLocator.C

```diff
@@ -138,9 +137,25 @@ NearestNodeLocator::findNodes()
     const std::map<dof_id_type, std::vector<dof_id_type>> & node_to_elem_map =
         _mesh.nodeToElemMap();
 
+    // Use KdTrees
+    std::vector<Point> master_points(trial_master_nodes.size());
+    for (unsigned int i = 0; i < trial_master_nodes.size(); ++i) {
+      const Node &node = _mesh.nodeRef(trial_master_nodes[i]);
+      for (unsigned int j = 0; j < 3; ++j)
+        master_points[i](j) = node(j);
+    }
+
+    PointListAdaptor<3> point_list_adaptor(master_points);
+
+    kd_tree_t my_kd_tree(3, point_list_adaptor,
+                         nanoflann::KDTreeSingleIndexAdaptorParams(10));
```

Comment:
> This is the max leaf size for the kd_tree. Should the user be allowed to set this parameter with the default value being 10?

That is a way how to fix the magic constant if you want to give users that power.